ggml-cuda.cu

```diff
@@ -5987,6 +5987,88 @@ static  __global__ void im2col_f32_f16(
     }
 }
 
+#define CUDA_FLASH_ATTENTION_BLOCK_SIZE 256
+
+template<int block_size>
+static __global__ void flash_attn_f32(const float* q, const float* k,const float* v, float* dst, float kq_scale,
```

Comment:
Add `__restrict__` to pointers, see https://github.com/ggerganov/llama.cpp/pull/2140 . An Ampere I saw comparatively little difference but adding it shouldn't hurt.